test_conformance/integer_ops/test_extended_bit_ops_extract.cpp

```diff
@@ -0,0 +1,287 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <algorithm>
+#include <numeric>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include "procs.h"
+#include "harness/integer_ops_test_info.h"
+#include "harness/testHarness.h"
+
+template <typename T>
+static typename std::make_unsigned<T>::type
+arithmetic_shift_right(T tx, cl_uint count)
+{
+    typedef typename std::make_unsigned<T>::type unsigned_t;
+    unsigned_t x = static_cast<unsigned_t>(tx);
+
+    // To implement an arithmetic shift right:
+    // - If the sign bit is not set, shift as usual.
+    // - Otherwise, flip all of the bits, shift, then flip back.
+    unsigned_t s = -(x >> (sizeof(x) * 8 - 1));
+    unsigned_t result = (s ^ x) >> count ^ s;
+
+    return result;
+}
+
+template <typename T>
+static typename std::make_unsigned<T>::type
+cpu_bit_extract_signed(T tbase, cl_uint offset, cl_uint count)
+{
+    typedef typename std::make_signed<T>::type unsigned_t;
```

Comment:
Hi, @bashbaug , std::make_signed<T>::type does not reflect the literal meaning of unsigned_t, is there just a typo? Thanks very much for your comments.

Reply:
Let's move discussion to https://github.com/KhronosGroup/OpenCL-CTS/pull/1913.

It's been a while but I think this is a typo and I intended to use unsigned types exclusively.